dspace-api/src/main/java/org/dspace/authenticate/SAMLAuthentication.java

```diff
@@ -0,0 +1,143 @@
+package org.dspace.authenticate;
+
+import org.apache.commons.collections.ListUtils;
+import org.apache.log4j.Logger;
+import org.dspace.authenticate.factory.AuthenticateServiceFactory;
+import org.dspace.authenticate.service.AuthenticationService;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.Group;
+import org.dspace.eperson.factory.EPersonServiceFactory;
+import org.dspace.eperson.service.EPersonService;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.saml.SAMLCredential;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.sql.SQLException;
+import java.util.List;
+
+public class SAMLAuthentication implements AuthenticationMethod {
+
+    /** log4j category */
+    private static Logger log = Logger.getLogger(SAMLAuthentication.class);
+
+    private final transient AuthenticationService authenticationService
+            = AuthenticateServiceFactory.getInstance().getAuthenticationService();
+    protected EPersonService ePersonService = EPersonServiceFactory.getInstance().getEPersonService();
+
+    @Override
+    public boolean canSelfRegister(Context context,
+                                   HttpServletRequest request,
+                                   String username)
+        throws SQLException {
+        return true;
+    }
+
+    @Override
+    public void initEPerson(Context context,
+                            HttpServletRequest request,
+                            EPerson eperson)
+        throws SQLException {
+        // We don't do anything because all our work is done in authenticate
```

Comment:
I don't think so because I followed LDAPAuthentication, PasswordAuthentication, and ShibAuthentication for guidance and `initEperson()` isn't being used in those classes as well.